feed-rs/src/xml/mod.rs

```diff
@@ -259,18 +257,27 @@ impl<R: BufRead> SourceState<R> {
 
     // Returns the next event
     fn fetch_next(&mut self) -> XmlResult<Option<XmlEvent>> {
+        let decoder = self.reader.decoder();
         let reader = &mut self.reader;
+
         loop {
-            let (ns, event) = reader.read_namespaced_event(&mut self.buf_event, &mut self.buf_ns)?;
+            let (ns_resolution, event) = reader.read_resolved_event_into(&mut self.buf_event)?;
 
             match event {
                 // Start of an element
                 Event::Start(ref e) => {
                     // Parse the namespace
-                    let namespace = ns
-                        .map(|bytes| reader.decode(bytes))
-                        .map(|s| NS::parse(s.as_ref()))
-                        .unwrap_or(self.default_namespace);
+                    let namespace = match ns_resolution {
```

Comment:
I don't know enough about XML to determine whether `ResolveResult::Bound` and `ResolveResult::Unknown` require a different treatment.

Reply:
I think its best to map an unknown namespace (i.e. a prefix that doesn't map to a known fully-qualified name) to the default space (since we can't really do anything with it, and will fail trying to parse it)